### PR TITLE
Refactor(plugins): Move jinja filter code for arista.avd.status_render to PyAVD

### PR DESCRIPTION
**Pull request by Shivani-gslab**
## Change Summary

Move jinja filter code for arista.avd.status_render to PyAVD

## Related Issue(s)

Fixes #<ISSUE ID>

## Component(s) name
Filter

- `arista.avd.status_render`

## Proposed changes
Move jinja filter code and tests for arista.avd.status_render to PyAVD

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/plugins/filter/markdown_rendering.py [hunk 1]
```diff
@@ -5,6 +5,21 @@
...
 __metaclass__ = type
 
+from ansible.errors import AnsibleFilterError
+
+from ansible_collections.arista.avd.plugins.plugin_utils.pyavd_wrappers import RaiseOnUse, wrap_filter
+
+PLUGIN_NAME = "arista.avd.status_render"
+
+try:
+    from pyavd.j2filters.markdown_rendering import status_render
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/filter/markdown_rendering.py]: Let's fix the names to be consistent `markdown_rendering`

**Review by ClausHolbechArista — COMMENTED** [review]
The title of the PR looks wrong :)

File: ansible_collections/arista/avd/plugins/filter/markdown_rendering.py (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/filter/markdown_rendering.py]: ah my bad. The filter name is actually `status_render` so we should use that everywhere - also for filenames.

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/plugins/filter/markdown_rendering.py]: `status_render` to `markdown_rendering` ?

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/plugins/filter/markdown_rendering.py]: Okay, lemme change the file name

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/plugins/filter/markdown_rendering.py]: done

**Review by ClausHolbechArista — APPROVED** [review]

File: python-avd/tests/pyavd/j2filters/test_status_render.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
```
Comment (MaheshGSLAB) [inline python-avd/tests/pyavd/j2filters/test_status_render.py]:
```suggestion
from __future__ import annotations
```

File: python-avd/tests/pyavd/j2filters/test_status_render.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+import pytest
+from pyavd.j2filters.status_render import status_render
+
+STATE_STRINGS = [("PASS", "github", ":white_check_mark:"), ("FAIL", "test", "FAIL")]
```
Comment (gmuloc) [inline python-avd/tests/pyavd/j2filters/test_status_render.py]: let's add a test to valdiate the `upper` part of the Github code
```suggestion
STATE_STRINGS = [("PASS", "github", ":white_check_mark:"), ("fail", "github", ":x:"), ("FAIL", "test", "FAIL")]
```

**Review by MaheshGSLAB — APPROVED** [review]

**Review by gmuloc — APPROVED** [review]

**Pull request merged**